beetsplug/permissions.py

```diff
@@ -28,13 +30,43 @@ def check_permissions(path, permission):
     return oct(os.stat(path).st_mode & 0o777) == oct(permission)
 
 
+def get_music_directories(music_directory, imported_item):
```

Comment:
We already do a very similar thing in `beets.util.prune_dirs` using `ancestry` (the first couple of stanzas, which produce `ancestors`). Can we reuse that code instead of duplicating the logic?